qiskit/providers/models/backendproperties.py

```diff
@@ -122,8 +126,144 @@ def __init__(self, backend_name, backend_version, last_update_date,
+        self.general = general
         self.qubits = qubits
         self.gates = gates
-        self.general = general
+
+        self._qubits = defaultdict(dict)
+        for qubit, props in enumerate(qubits):
+            formatted_props = {}
+            for prop in props:
+                value = self._apply_prefix(prop.value, prop.unit)
+                formatted_props[prop.name] = (value, prop.date)
+                self._qubits[qubit] = formatted_props
+
+        self._gates = defaultdict(dict)
+        for gate in gates:
+            qubits = _to_tuple(gate.qubits)
+            formatted_props = {}
+            formatted_props['name'] = gate.name
+            for param in gate.parameters:
+                value = self._apply_prefix(param.value, param.unit)
+                formatted_props[param.name] = (value, param.date)
+            self._gates[gate.gate][qubits] = formatted_props
 
         super().__init__(**kwargs)
+
+    def gate_error(self, operation: str, qubits: Union[int, Iterable[int]]):
+        """
+        Return gate error estimates from backend properties.
+
+        Args:
+            operation (str): The operation for which to get the error.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        return self.gate_property(operation, qubits, 'gate_error')[0] # Throw away datetime at index 1
+
+    def gate_length(self, operation: str, qubits: Union[int, Iterable[int]]):
+        """
+        Return the duration of the gate in units of seconds.
+
+        Args:
+            operation (str): The operation for which to get the duration.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        return self.gate_property(operation, qubits, 'gate_length')[0] # Throw away datetime at index 1
+
+    def gate_property(self,
+                      operation: str,
+                      qubits: Union[int, Iterable[int]] = None,
+                      name: str = None) -> Tuple[Any, datetime.datetime]:
+        """
+        Return the gate properties of the given qubit and property.
+
+        Args:
+            operation (str) : Name of the gate
+            qubits (Union[int, Iterable[int]]): The property to look for.
+            name (str): Optionally used to specify within the heirarchy which
+            property to return.
+
+        Return:
+            Gate properties of the given qubit and property, if it was given by the
+            backend, otherwise, return `None`.
+
+        Raises:
+	        PulseError: If `qubits` is `None` and `name` is `not None` or
+            if the property is not found.
+        """
+        result = self._gates
+        try:
+            if operation is not None:
```

Comment:
Remove this if statement, the user will provide `operation`

Reply:
This is an example of the user shooting themselves in the foot. We can do our best to help them, but they can still do useless things. Luckily, they'll get the same error as they'd get if they passed any other missing key, just a `KeyError`